src/UnkObjects/unk8600F8.cpp

```diff
@@ -25,3 +26,214 @@ void unk_8600F8::loc_611DC0() {
     *x = this->unk14;
     this->unk14 = nullptr;
 }
+
+unk_612150 *unk_8600F8::sub_6124A0(const char *a) {
+    if (this->entries == NULL) {
+        return nullptr;
+    }
+
+    int index = get_entry_index(a);
+    
+    if (this->entries[index] == nullptr) {
+        return nullptr;
+    }
+
+    hash_map_entry *edi = this->entries[index];
+    
+    while (1) {
+        char *entryName = edi->name;
+
+        const char* a_p = a;
+        while (1) {
```

Comment:
Also, again about typing, `while (true)`. I realize ints and bools are interchangeable in C, but I like to keep my C++ code more type-explicit.